Make animation editor change tracks positions instead of swapping them

YeldhamDev opened the pull request:
In the animation editor dragging and dropping a track doesn't change its position as one would expect, instead, it swaps around the tracks. This commit changes that, by creating a new method: `track_move_to()`, and making use of it there.

Closes #27288.

YeldhamDev committed:
Make animation editor change tracks positions instead of swapping them

avencherus commented:
Have a build running, I'm thinking it is related to this:  https://github.com/godotengine/godot/issues/27288

avencherus commented:
So it is aimed at fixing that.  In the flat view it seems to be working better, though it looks like it drops itself down one row below where it was dropped at times.  Maybe its still trying to respect their groupings?

![ezgif com-video-to-gif (4)](https://user-images.githubusercontent.com/13004169/58378023-5fc2cc80-7f94-11e9-9532-546f1db632d9.gif)

In the grouped view, I still have no idea what exactly is going on or what is supposed to happen. 😕  The groups seem to be not moveable in a predictable manner.  The tracks can be dragged around and outside of the groups...  Perhaps a separate issue.

![ezgif com-video-to-gif (1)](https://user-images.githubusercontent.com/13004169/58378027-76692380-7f94-11e9-9727-682c4a5040f5.gif)

YeldhamDev commented:
@avencherus Thank you very much for helping testing my commits once again!

I changed how tracks are moved while grouped. Now, they can only be moved between tracks of their group. Nothing will happen will dragging them outside of it, not even highlighting.

And about the tracks being moved incorrectly, fixed. Let's say that working with a structure that changes its content's positions when moving things around is kinda of a PITA.

avencherus commented:
> And about the tracks being moved incorrectly, fixed. Let's say that working with a structure that changes its content's positions when moving things around is kinda of a PITA.

😆 

Least I can do, I just very much appreciate your contributions in these areas.  The AnimationPlayer is near and dear to me and my partner, as all these "smaller" things have been untold hours of frustration and lost time to us in our development.  So you have our combined thanks.

I took a look at your updates.  I like the blocking icon on the drag over groups, that's more clear in my opinion, and removes confusion.

There was an issue going on though with the selection highlight.

When in group view, sometimes when re-arranging the tracks, the highlight jumps to another group.  Without more knowledge of it, I can't really discern how to reproduce it, but here is a video of it happening.

![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/13004169/58397119-46835400-8058-11e9-8e69-c985fa474f38.gif)

The second piece of it, is that when in the flat view, the highlight doesn't follow to the new location.  In a big track list with similar properties, I found it difficult to locate after things re-arrange themselves.  Here is a video of dragging down one slot, and then dropping it down further away.  The highlight stays at the index where the moved track was originally.

![ezgif com-video-to-gif(3)](https://user-images.githubusercontent.com/13004169/58397192-9235fd80-8058-11e9-91a5-2c06835a5ced.gif)

YeldhamDev commented:
Tracks are now highlighted when moved, and the wrong group highlighting should be fixed.

avencherus commented:
Working nicely.  🎉

akien-mga commented:
Thanks!
